### ISSUE TITLE
[Training] Batchnormalisation supported?

### ISSUE THREAD

**Issue by pauldog** [microsoft/onnxruntime#17879]
### Describe the issue

When trying to get the training model with a batchnormalisation node I get the error:

failed to find NodeArg by name f2_bn2.running_var_grad

Is batchnormalisation not supported for training?

![snapshot5](https://github.com/microsoft/onnxruntime/assets/33497043/da2f2afd-1797-4005-9172-027a41c5e344)

This is a trained model converted from here: https://github.com/suragnair/alpha-zero-general (othello)

### To reproduce

Create a training artefact with a batch normalisation node.

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.15.1

### PyTorch Version

1.13.1

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA 1.13.1

**Comment by baijumeswani** [comment]
> failed to find NodeArg by name f2_bn2.running_var_grad

the BatchNorm training inputs (running mean and running var) are not trainable. They are initializers in the model that get updated, but they are not trainable.

You will have to put the initializers in the `frozen_params` list while generating the training artifacts:

```py
frozen_params = ["f2_bn2.running_var", f2_bn2.running_mean"] # and potentially other similar initializers
```

**Comment by pauldog** [comment]
Thanks. Thought that might be the problem. However now I get the error:

![snapshot5](https://github.com/microsoft/onnxruntime/assets/33497043/db618cab-abf3-4046-8be1-4d5672ad54d0)

**Comment by baijumeswani** [comment]
Could you share the forward onnx model? Will try to find the reason for the issue.

**Comment by baijumeswani** [comment]
Closing this issue as I am not able to reproduce the error. If you're still encountering this, please reopen the issue.
